### ISSUE TITLE
remove this var "&autoplay=0"

### ISSUE THREAD

**Issue by rezad1393** [SimonBrazell/privacy-redirect#162]
when I open a youtube link this gets added to end of it.
I have disabled autoplay in firefox and dont need this parameter but there is no way to remove it from addon.
I only can set "&autoplay=0" or "&autoplay=1" which the second one doesn't do anything as I disabled autoplay globally.

I think this is a overlooked (kinda) bug in addon that because it is a flip choice between on and off for autoplay it always adds this parameter to end of redirected url.
maybe add a third option that doesn't touch autoplay set by firefox?

**Comment by SimonBrazell** [comment]
I think it depends what the default behaviour of invidious is, if it's `autoplay` disabled then we should be safe to remove `&autoplay=0` completely.

**Comment by rezad1393** [comment]
I think if you add a parameter to the url that user cant remove whatever he does, it is not a good thing.
and to be precise, this just happens with redirects from youtube to invidious.
also this also happens with channels and users too like this:
https://invidious.kavin.rocks/channel/UCldfgbzNILYZA4dmDt4Cd6A?autoplay=0
which is even more not-needed.

**Comment by TheFrenchGhosty** [comment]
> I think it depends what the default behaviour of invidious is, if it's `autoplay` disabled then we should be safe to remove `&autoplay=0` completely.

From what I know, it is.

**Comment by rezad1393** [comment]
> @SimonBrazell
> 
> > I think it depends what the default behaviour of invidious is, if it's `autoplay` disabled then we should be safe to remove `&autoplay=0` completely.
> 
> From what I know, it is.

you mean it is safe or you mean that it is removed?
because it is not removed.

**Comment by SimonBrazell** [comment]
> you mean it is safe or you mean that it is removed?
because it is not removed.

I mean there should be some way for users to turn autoplay off via the extension unless it is off by default, if its the default then I'm happy to just remove the `autoplay=0` param altogether.

**Comment by rezad1393** [comment]
> > you mean it is safe or you mean that it is removed?
> > because it is not removed.
> 
> I mean there should be some way for users to turn autoplay off via the extension unless it is off by default, if its the default then I'm happy to just remove the `autoplay=0` param altogether.

you are the developer?
if so then maybe add a third option that doesnt change the autoplay.
so user has 3 option: &autoplay=0 on, &autoplay=1 and no parameter added for autoplay.

also can you fix it so that it just added only to youtube video page and not channel or search page.

**Comment by SimonBrazell** [comment]
> you are the developer?

Yes.

I'll think about this next time I work on the project, probably in the next week or two, I'll let you know what I come up with.

PRs are welcome too if you wanna have a go at it.

**Comment by rezad1393** [comment]
> > you are the developer?
> 
> Yes.
> 
> I'll think about this next time I work on the project, probably in the next week or two, I'll let you know what I come up with.
> 
> PRs are welcome too if you wanna have a go at it.

I am too dumb to do computer programming.

**Comment by SimonBrazell** [comment]
> I am too dumb to do computer programming.

I'm sure that's not the case, its like everything, the hardest part is probably just getting motivated to start. 🙂

**Comment by rezad1393** [comment]
> > I am too dumb to do computer programming.
> 
> I'm sure that's not the case, its like everything, the hardest part is probably just getting motivated to start. slightly_smiling_face

many years ago I wanted to learn firefox add writing and firefox went and change the whole system.
that was a message from creator telling I was going stray. :)

but thanks for the answers.
I will wait for your update.
have a nice day.

**Comment by SimonBrazell** [comment]
Haha yeah they are probably about to change it all again now then... 🙂 even if they did though everything you learned would be applicable to any web tech as the project is mostly just plain JS, HTML and CSS.

All good, enjoy the rest of your weekend!

**Comment by rezad1393** [comment]
> Haha yeah they are probably about to change it all again now then... slightly_smiling_face even if they did though everything you learned would be applicable to any web tech as the project is mostly just plain JS, HTML and CSS.
> 
> All good, enjoy the rest of your weekend!

yes I am trying to learn some javascipt to do some custom user-script for the website I frequently visit.
have a nice weekend too (too in my country it is start of week).

**Comment by SimonBrazell** [comment]
> yes I am trying to learn some javascipt to do some custom user-script for the website I frequently visit.

Good luck! If you wanted to have a go at some dev on this project I would give you advice etc. if you ran into difficulty.

> even though in my country it is start of week

Ah damn sorry, here in AU its kinda a long weekend.

**Comment by rezad1393** [comment]
> > yes I am trying to learn some javascipt to do some custom user-script for the website I frequently visit.
> 
> Good luck! If you wanted to have a go at some dev on this project I would give you advice etc. if you ran into difficulty.
> 
> > even though in my country it is start of week
> 
> Ah damn sorry, here in AU its kinda a long weekend.

maybe a simple step by step javascript guide but not for front-end development ,just for user-script.
I trying to find that but they are either font-end focused that also ties to backend, or user-scripts guide that assume you know medium level javascript.

also no worry it not because of geography. I live in Iran so the start of week is Saturday.

**Comment by SimonBrazell** [comment]
> maybe a simple step by step javascript guide but not for front-end development ,just for user-script.
I trying to find that but they are either font-end focused that also ties to backend, or user-scripts guide that assume you know medium level javascript.

I learned programming fundamentals first with python and java, then learned JS while building web apps, I recently tried to teach a friend JS and it quickly became apparent that they needed to learn programing basics first so maybe that's a good place to start instead?

> also no worry it is not because of geography. I live in Iran so the start of week is Saturday.

Haha had me confused for a sec, was thinking, "but it must be the weekend everywhere"... shows how much I know. 🙂

**Comment by rezad1393** [comment]
> > maybe a simple step by step javascript guide but not for front-end development ,just for user-script.
> > I trying to find that but they are either font-end focused that also ties to backend, or user-scripts guide that assume you know medium level javascript.
> 
> I learned programming fundamentals first with python and java, then learned JS while building web apps, I recently tried to teach a friend JS and it quickly became apparent that they needed to learn programing basics first so maybe that's a good place to start instead?
> 
> > also no worry it is not because of geography. I live in Iran so the start of week is Saturday.
> 
> Haha had me confused for a sec, was thinking, "but it must be the weekend everywhere"... shows how much I know. slightly_smiling_face

so you mean python learning is better for starter?
I had my eyes on this :
[https://automatetheboringstuff.com/2e/chapter0/](url)
is this too basic for to jumpstart to other program languages because it seems to be good for layman but it seems to skip fundamentals.

**Comment by Mennaruuk** [comment]
I think it'd be helpful to put some kind of note under the autoplay parameter reminding users to disable autoplay for whatever Invidious instance they're on. As OP mentioned, even if autoplay is disabled in the Privacy Redirect addon, Firefox would block it anyway by default, and that's been the case since March 2019 (at least audio blocking*). So there's an extra step that users have to take in order to truly have autoplay work for Invidious instances. I also noticed that this only applies to YouTube links. When I directly enter an Invidious link into Firefox, the addon doesn't add the autoplay parameter to it, which is most likely due to the nature of the addon only redirecting YouTube links to Invidious links.

*When Firefox blocks audio only, the video still doesn't start. So what does "block audio only" mean? Does it mean the video still gets downloaded/streamed but it won't start until user plays it?

https://user-images.githubusercontent.com/52135169/106372769-0a31ec00-6341-11eb-8858-bc486e1d4462.mp4

**Comment by SimonBrazell** [comment]
> so you mean python learning is better for starter?

That's how I started, I think it's usually picked because the syntax is closer to written language and lack of clutter so it's easier to pick up. But honestly you can probably learn with any language, once you know the fundamentals you are set.

My partner is currently trying to learn with [https://www.freecodecamp.org/]( https://www.freecodecamp.org), seems to be having some success with it, although it seems more focused on web technologies but I think there is some python courses in there, if you haven't tried it already maybe start from the start and see how you go?

> When I directly enter an Invidious link into Firefox, the addon doesn't add the autoplay parameter to it, which is most likely due to the nature of the addon only redirecting YouTube links to Invidious links.

Yes the addon doesn't touch your request if it's not to YT so there won't be any params added to the URL. However there is an advanced option to include your Invidious settings in a cookie that Invidious instances recognise if you want to give this a try? You'll need to whitelist any invidious sites you use if you use a cookie blocker.

**Comment by rezad1393** [comment]
my issue is not the autoplay or any other setting.
but I tried it (cookie mode) anyway.
first it messes with the invidius setting. now does this even when I am using invidious instance and not redirecting from youtube.
so my settings are gone and also addon setting are not applied when not redirecting.
so the instance goes to its default setting.
and whenever I change the setting to my liking, some of them get reset again to default.
like the player style gets reset invidious, even though I set it to youtube in addin and then when addon doesnt do that I set it in option of instance but when I press the save the prefrences button and the website goes back to video page it is again in invidious style and not youtube.
same thing with " Show annotations by default: " and new "Preferred dash video quality: " setting.

but all of this is another bug that occur with cookie mode and it is not the main issue of me but I wanted to report them so that you know that they are there.
also when I set the cookie mode I still get the parameters and I saw that when I set other parameters I get them too in url.

like this:
&local=true&quality=dash&dark_mode=true&player_style=youtube&autoplay=1

but fortunately when I toggle them again to their off mode they are removed from the url.
all except the autoplay one which is the one I mentioned here.

so to recap cookie mode is buggy but that is not my issue.
again my issue is that your addon doesnt seem to have a default mode as in ,dont touch it mode, for autoplay.

please add that.
or define not-autoplay as default and dont add parameter for that when set to not autoplay and only add when user enables autoplay. this is easier and also more sane option.

**Comment by rezad1393** [comment]
about the main bug you only need to change this (I think, as I said I am a noob)
  url.searchParams.append("autoplay", invidiousAutoplay ? 1 : 0);

so that it doesnt append anything it is set to zero

**Comment by opusforlife2** [comment]
@SimonBrazell The parameter is being added even to playlist URLs. I think it is better to remove it and release a minor update.
